packages/eas-cli/src/build/build.ts

```diff
@@ -284,15 +311,26 @@ async function uploadProjectAsync<TPlatform extends Platform>(
       }
     );
   } catch (err: any) {
-    let errMessage = 'Failed to upload the project tarball to EAS Build';
+    let errMessage = bucketKey
+      ? 'Failed to upload metadata to EAS Build'
+      : 'Failed to upload the project tarball to EAS Build';
```

Comment:
To me, managing state by existence of `bucketKey` feels too clever.

I would suggest to extract metadata upload to a separate function `uploadMetadataAsync`. This would also remove the inclination to have the same `metadataLocation` in all upload types, which, [as discussed](https://github.com/expo/eas-build/pull/301#discussion_r1426579106), is a little confusing next to `bucketKey`. `bucketKey` is clearly to-be-appended-to-bucket-URL. `metadataLocation` is… what?

Reply:
Maybe `metadataBucketKey`? What do you think?

Reply:
@khamilowicz what did you mean by resolving this conversation? Have you talked to @szdziedzic and convinced him using `metadataLocation` is better?